### PR TITLE
Refactor/DEVTOOLING-191 and DEVTOOLING-646

### PR DESCRIPTION
**Pull request by sureshperiyappan**

### CODE REVIEW

**Comment by carnellj-genesys** [comment]
Need to resolve merge conflicts

**Review by dginty4 — APPROVED** [review]

**Review by carnellj-genesys — APPROVED** [review]

File: genesyscloud/routing_wrapupcode/data_source_genesyscloud_routing_wrapupcode.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceRoutingWrapupcodeRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	sdkConfig := m.(*provider.ProviderMeta).ClientConfig
+	proxy := newRoutingWrapupcodeProxy(sdkConfig)
```
Comment (charliecon) [inline genesyscloud/routing_wrapupcode/data_source_genesyscloud_routing_wrapupcode.go:19]: `newRoutingWrapupcodeProxy` -> `getRoutingWrapupcodeProxy`

`newRoutingWrapupcodeProxy` should only be called from inside `getRoutingWrapupcodeProxy`

File: main.go [hunk 2]
```diff
@@ -147,7 +149,7 @@ func main() {
 
 	if debugMode {
 		opts.Debug = true
-		opts.ProviderAddr = "genesys.com/mypurecloud/genesyscloud"
+		opts.ProviderAddr = "registry.terraform.io/mypurecloud/genesyscloud"
 	}
```
Comment (charliecon) [inline main.go:153]: This was just changed to `genesys.com` by Brian in [this PR](https://github.com/MyPureCloud/terraform-provider-genesyscloud/commit/a96cdae3f2548e9b0d1a58249214a2534c3542f9). You might need to merge dev into this branch.

File: genesyscloud/routing_wrapupcode/data_source_genesyscloud_routing_wrapupcode.go (hunk 1 above)
Comment (sureshperiyappan) [reply genesyscloud/routing_wrapupcode/data_source_genesyscloud_routing_wrapupcode.go:19]: Thanks for catching this :) very bad coding

**Comment by sureshperiyappan** [comment]
@charliecon update the PR please have look at it

**Review by charliecon — APPROVED** [review]

**Pull request merged**